### ISSUE TITLE
RedirectViews failing to sync with S3 with boto3

### ISSUE THREAD

**Issue by palewire** [datadesk/django-bakery#102]
```python
uploading /home/travis/build/datadesk/bigbuilder/.build/boto3/projects/sitemap.xml
uploading /home/travis/build/datadesk/bigbuilder/.build/boto3/projects/la-pol-ca-34th-congressional-issues/index.html
uploading /home/travis/build/datadesk/bigbuilder/.build/boto3/projects/mega-millions-lottery-simulator/index.html
uploading /home/travis/build/datadesk/bigbuilder/.build/404.html
uploading /home/travis/build/datadesk/bigbuilder/.build/boto3/projects/feeds/latest.xml
uploading /home/travis/build/datadesk/bigbuilder/.build/index.html
uploading /home/travis/build/datadesk/bigbuilder/.build/boto3/projects/grade-the-president/index.html
uploading /home/travis/build/datadesk/bigbuilder/.build/boto3/projects/la-me-trump-immigrant-stories/index.html
uploading /home/travis/build/datadesk/bigbuilder/.build/boto3/projects/google-news-sitemap.xml
uploading /home/travis/build/datadesk/bigbuilder/.build/boto3/projects/index.html
uploading /home/travis/build/datadesk/bigbuilder/.build/boto3/projects/la-na-pol-first-presidential-debate-primer/index.html
Adding S3 redirect header from boto3/projects/mega-millions-lottery-simulator/index.html to /boto3/projects/la-g-mega-millions-lottery-simulator
Traceback (most recent call last):
  File "manage.py", line 8, in <module>
    execute_from_command_line(sys.argv)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/django/core/management/__init__.py", line 367, in execute_from_command_line
    utility.execute()
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/django/core/management/__init__.py", line 359, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/django/core/management/base.py", line 294, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/django/core/management/base.py", line 345, in execute
    output = self.handle(*args, **options)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/bigbuild/management/commands/publish.py", line 14, in handle
    super(Command, self).handle(*args, **options)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/bakery/management/commands/publish.py", line 125, in handle
    getattr(view, 'post_publish')(self.bucket)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/bakery/views/base.py", line 201, in post_publish
    key = bucket.get_key(self.build_path)
AttributeError: 's3.Bucket' object has no attribute 'get_key'
```

**Comment by palewire** [comment]
It looks like the problem is in the old boto API calls in the post_publish method.

```python
    def post_publish(self, bucket):
        logger.debug("Adding S3 redirect header from %s to %s" % (
            self.build_path,
            self.get_redirect_url()
        ))
        key = bucket.get_key(self.build_path)
        key.copy(
            key.bucket,
            key.name,
            preserve_acl=True,
            metadata={'Content-Type': 'text/html'}
        )
        key.set_redirect(self.get_redirect_url())
        key.make_public()
```

One more mole for us to whack here @sheats.

**Comment by sheats** [comment]
Cool... I try to get a test and a fix for that soon!

**Comment by palewire** [comment]
I was unable to find an equal to ``set_redirect`` in the boto3 docs so I'm going to remove this broken code for the time being and do a release without it. The HTML meta data redirects will still be in place until we find a solution.

I also posted over in the boto3 issues pool to see if they could point us in the right direction. 

https://github.com/boto/boto3/issues/1047

**Comment by sheats** [comment]
Looks like what we are after is the WebsiteRedirectLocation arg to http://boto3.readthedocs.io/en/latest/reference/services/s3.html#S3.Client.copy_object or similar method call.  

Doesn't seem like you can pass it in during the upload_file call so seems we might have to follow the same strategy as you guys did before by copying the object and setting this attribute.

**Comment by palewire** [comment]
I see. Good find!

**Comment by sheats** [comment]
I'm not sure when I'm going to have time to implement this so wanted to put what I learned tonight here in case someone else wants to do it.

Looks like we basically need a call like this:
```
s3_client.copy_object(
    ACL='public-read', 
    Bucket='mock_bucket', 
    CopySource={
         'Bucket': 'mock_bucket', 
         'Key': self.build_path}, 
     Key=self.build_path, 
     WebsiteRedirectLocation=self.get_redirect_url())
```

It seems like we will need to pass the s3_client into the post_publish method from the publish script.

And I can't seem to figure out how to access the bucket name from the bucket object that is passed into the method so if that's not possible we'll probably need to pass it in as a string.

**Comment by palewire** [comment]
I think I patched this and just shipped v0.9.3.
